server/api/api.go

```diff
@@ -1443,6 +1448,250 @@ func (a *API) getWorkspaceUsers(w http.ResponseWriter, r *http.Request) {
 	auditRec.Success()
 }
 
+// subscriptions
+
+func (a *API) handleCreateSubscription(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation POST /api/v1/workspaces/{workspaceID}/subscriptions createSubscription
+	//
+	// Creates a subscription to a block for a user. The user will receive change notifications for the block.
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: workspaceID
+	//   in: path
+	//   description: Workspace ID
+	//   required: true
+	//   type: string
+	// - name: Body
+	//   in: body
+	//   description: subscription definition
+	//   required: true
+	//   schema:
+	//     "$ref": "#/definitions/Subscription"
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//         "$ref": "#/definitions/User"
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	container, err := a.getContainer(r)
+	if err != nil {
+		a.noContainerErrorResponse(w, r.URL.Path, err)
+		return
+	}
+
+	requestBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	var sub model.Subscription
+
+	err = json.Unmarshal(requestBody, &sub)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	if err = sub.IsValid(); err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusBadRequest, "", err)
+	}
+
+	ctx := r.Context()
+	session := ctx.Value(sessionContextKey).(*model.Session)
+
+	auditRec := a.makeAuditRecord(r, "createSubscription", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelModify, auditRec)
+	auditRec.AddMeta("user_id", sub.SubscriberID)
+	auditRec.AddMeta("block_id", sub.BlockID)
+
+	// User can only create subscriptions for themselves (for now)
+	if session.UserID != sub.SubscriberID {
+		a.errorResponse(w, r.URL.Path, http.StatusBadRequest, "userID and subscriberID mismatch", nil)
+		return
+	}
+
+	subNew, err := a.app.CreateSubscription(*container, &sub)
```

Comment:
Should we verify if you can create a subscription for a non valid block (I guess we are checking the permissions on the notifications, but maybe makes sense to check the permissions also here, or maybe not).

Reply:
Good point. I'll add a check for valid block ID.

Reply:
Fixed via https://github.com/wiggin77/focalboard/commit/4a4a1b5d6afa918a65c8a34d59ad93a7029a3300